spec.bs

```diff
@@ -309,26 +401,242 @@ maps to an internal [=fenced frame config=] [=struct=].
+
+  1. <span class=XXX>TODO: Fill this in</span>
+</div>
+
+An <dfn export for=fencedframetype>exfiltration budget metadata</dfn> is a struct containing an
+<dfn export for="exfiltration budget metadata">origin</dfn>, which is an [=origin=]; and an <dfn
+export for="exfiltration budget metadata">amount to debit</dfn>, which is a non-negative valid
+floating point number <span class=XXX>TODO: ref to float</span>.
+
+An <dfn export for=fencedframetype>exfiltration budget metadata reference</dfn> is a struct
+containing an <dfn export for="exfiltration budget metadata reference">origin</dfn>, which is an
+[=origin=]; and an <dfn export for="exfiltration budget metadata reference">amount to debit
+reference</dfn>, which is a mutable reference to a non-negative valid floating point number.
+<span class=XXX>TODO: are mutable references a thing in spec?</span>
+
+An <dfn export for=fencedframetype>embedder shared storage context</dfn> is a string.
+
+A <dfn export>partition nonce</dfn> is <span class=XXX>TODO: Specify the type for this.</span>
 
 A <dfn export>fenced frame config</dfn> is a struct with the following [=struct/items=]:
 
-* An <dfn for="fenced frame config">urn uuid</dfn>, a [=urn uuid=] string
-* <span class=XXX>TODO: Specify the other members</span>
+<dl export dfn-for="fenced frame config">
+  : <dfn>mapped url</dfn>
+  :: null, or a struct containing a [=fencedframetype/mapped url=] and a
+     [=visibility/visibility to embedder=]
+
+  : <dfn>container size</dfn>
+  :: null, or a struct containing a [=fencedframetype/size=] and a
+     [=visibility/visibility to content=]
+
+  : <dfn>content size</dfn>
+  :: null, or a struct containing a [=fencedframetype/size=] and a
+     [=visibility/visibility to embedder=]
+
+  : <dfn>interest group descriptor</dfn>
+  :: null, or a struct containing an [=fencedframetype/interest group descriptor=], a
+     [=visibility/visibility to embedder=], and a [=visibility/visibility to content=]
+
+  : <dfn>effective sandbox flags</dfn>
+  :: null, or a struct containing an [=fencedframetype/exhaustive set of sandbox flags=] and a
+     [=visibility/visibility to embedder=]
+
+  : <dfn>effective permissions</dfn>
+  :: null, or a struct containing an [=fencedframetype/exhaustive set of permissions=], a
+     [=visibility/visibility to embedder=], and a [=visibility/visibility to content=]
+
+  : <dfn>fenced frame reporter</dfn>
+  :: null, or a struct containing a [=fencedframetype/fenced frame reporter=], a
+     [=visibility/visibility to embedder=], and a [=visibility/visibility to content=]
+
+  : <dfn>exfiltration budget metadata</dfn>
+  :: null, or a struct containing an [=fencedframetype/exfiltration budget metadata=], a
+     [=visibility/visibility to embedder=], and a [=visibility/visibility to content=]
+
+  : <dfn>nested configs</dfn>
+  :: null, or a struct containing a sequence of [=fenced frame configs=] and a
```

Comment:
Could we formalize this a bit? I read here that this is a struct-or-null, but in https://pr-preview.s3.amazonaws.com/WICG/fenced-frame/pull/67.html#store-nested-configs we try and iterate over `nested configs` which doesn't really make sense since this is a struct. It sounds like we want this to be:

`nested configs: {
    configs: map<urn, config>;
    visibility: visibility;
}`

in pseudo code? If that's right, let's give this a _member_ that is a map that we can explicitly reference when performing the iteration in the aforementioned link (and anywhere else, if applicable).